New example app: Trip Planner

ssimic2 opened the pull request:
Trip Planner is an example app where users can collaborate on trip planning. In our example we have hardcoded a single trip called "West Coast Trip", including 3 cities. For each itinerary stop, users can add notes and "to do" items (place to see, description etc). This example intends to educate on how FF collaboration can empower existing apps having their own data model and source of truth. This project is work in progress. In its final form it will address following topics:

1. Containers. Users should be clear how to reason about when and where to apply FF containers, how to share them etc.
2. Basic DDSes. Users should be clear how "optimistic" DDSes behave, and how to apply that knowledge in our sample app. We are starting with `SharedMap` and `SharedString`.
3. Modeling Collaboration. Users should be clear how to model more complex collaboration data. In our sample app here, we have collections where each item can have place, description & category. We want to discuss how we are splitting the model in a way where users do not inadvertently revert each other's changes, while still considering performance implications.
4. Audience. Users should be clear how to use `audience` to enhance collaboration experience.
5. Signals. Users should be clear on distinguishing persistent vs non-persistent data, and how each of them apply to various use cases.
6. FF Eventing. Users should be clear how to use events to bind FF data. We have existing examples that talk about eventing, but here we are looking into that same topic, but at a larger scale.
7. Source of Truth. In our sample app, the source of truth is separate from FF data. Users should be clear how two data sources can be synchronized.
8. Container States. Users should be clear how container state can be observed and managed.
9. Schema Upgrades. To keep pace with FF lib. changes, users should be clear how they can safely execute schema upgrades.
10. Recovery. Users should be clear how they can recover corrupted data.

![TripPlanner](https://user-images.githubusercontent.com/7103655/171675338-c5e4e37b-681a-497c-bb62-f4b0a5d70e0a.png)

Josmithr commented:
100% nitpick, but would it be better to call the app "our-trip-planner" instead of "my-trip-planner" since it's collaborative?

ssimic2 commented:
> 100% nitpick, but would it be better to call the app "our-trip-planner" instead of "my-trip-planner" since it's collaborative?

Love it! Renamed, and ready for re-review. 

Changed example to use shared upvote counter to initially better align with our getting-started example, before we start introducing more complex DDSes.

ssimic2 commented on my-trip-planner/README.md:
```diff
@@ -0,0 +1,53 @@
+# About
+
+My Trip Planner is an example app where users can collaborate on trip planning. In our example we have hardcoded a single trip called "West Coast Trip", including 3 cities. For each itinerary stop, users can add notes and "to do" items (place to see, description etc). This example intends to educate on how FF collaboration can empower existing apps having their own data model and source of truth. ...
```
Made the change!

ChumpChief commented:
Since FF 2.0 and switching to focus on SharedTree, I think this example probably either needs to be updated or reconsidered.  Closing this PR since there has not been recent activity.
